Задание 1.1. Базовые свойства массивов

Вы анализируете данные о заказах интернет-магазина.

Для каждого заказа известны:

• время доставки в минутах;

• стоимость заказа в рублях;

• количество товаров в заказе.

Создайте массивы:

import numpy as np

delivery_time = np.array([35, 42, 28, 51, 39, 63, 31, 45])

order_price = np.array([1200, 3500, 800, 4200, 1900, 5600,
1500, 2800])

items_count = np.array([2, 5, 1, 7, 3, 8, 2, 4])

Что необходимо сделать
1. Для каждого массива выведите:
◦ shape;
◦ ndim;
◦ size.
2. Объясните, что означает size в контексте рассматриваемых данных.

size — это количество наблюдений (заказов) в массиве. В данном случае в каждом массиве по 8 заказов.

3. Проверьте с помощью Python, что все три массива содержат одинаковое количество
наблюдений.

4. Объясните, почему каждый индекс в трёх массивах должен соответствовать одному и
тому же заказу.

Потому что массивы описывают разные характеристики одних и тех же объектов. Если порядок будет нарушен, мы получим несогласованные данные (например, время доставки от одного заказа, а стоимость — от другого). Это приведёт к неверным выводам.

Вопрос для анализа

Что произойдёт с результатами анализа, если delivery_time содержит 8 наблюдений, а
order_price — только 7? Почему это является проблемой не только с точки зрения
Python, но и с точки зрения качества данных?

Если delivery_time содержит 8 наблюдений, а order_price — 7, то:

Python не сможет корректно выполнять поэлементные операции (например, delivery_time + order_price вызовет ошибку).

С точки зрения качества данных это означает потерю или несогласованность информации: неизвестно, какому заказу соответствует пропущенная стоимость. Анализ становится ненадёжным.



In [ ]:
import numpy as np
delivery_time = np.array([35, 42, 28, 51, 39, 63, 31, 45])
order_price = np.array([1200, 3500, 800, 4200, 1900, 5600, 1500, 2800])
items_count = np.array([2, 5, 1, 7, 3, 8, 2, 4])
# 1. Вывод shape, ndim, size для каждого массива
for name, arr in [("delivery_time", delivery_time),
                  ("order_price", order_price),
                  ("items_count", items_count)]:
    print(f"{name}: shape={arr.shape}, ndim={arr.ndim}, size={arr.size}")
# 3. Проверка одинакового количества наблюдений
print("\nОдинаковое количество наблюдений:",
      delivery_time.size == order_price.size == items_count.size)

delivery_time: shape=(8,), ndim=1, size=8
order_price: shape=(8,), ndim=1, size=8
items_count: shape=(8,), ndim=1, size=8

Одинаковое количество наблюдений: True


Задание 1.2. Двумерные массивы

Объедините данные в единую матрицу orders.

Каждая строка должна соответствовать одному заказу, а столбцы должны содержать:

1. время доставки;

2. стоимость заказа;

3. количество товаров.

Создайте матрицу:

orders = np.array([

 [35, 1200, 2],

 [42, 3500, 5],

 [28, 800, 1],

 [51, 4200, 7],

 [39, 1900, 3],

 [63, 5600, 8],

 [31, 1500, 2],

 [45, 2800, 4]

])

Что необходимо сделать

1. Выведите orders.shape.

2. С помощью индексации получите:

◦ данные первого заказа;

◦ время доставки всех заказов;

◦ стоимость всех заказов;

◦ количество товаров в шестом заказе.

3. С помощью срезов получите:

◦ все заказы начиная с третьего;

◦ первые пять заказов и первые два признака.

Вопрос для анализа

Объясните, почему:

orders.shape
возвращает (8, 3), а:
orders.size
возвращает 24.

Что характеризует каждый из этих показателей?

orders.shape возвращает (8, 3), потому что в матрице 8 строк (заказов) и 3 столбца (признака).
orders.size возвращает 24, потому что это общее количество элементов: 8 × 3 = 24.

Что характеризует каждый показатель:

shape — структуру данных (сколько наблюдений и сколько признаков).

size — общий объём данных (суммарное число ячеек).

In [ ]:
orders = np.array([
    [35, 1200, 2],
    [42, 3500, 5],
    [28, 800, 1],
    [51, 4200, 7],
    [39, 1900, 3],
    [63, 5600, 8],
    [31, 1500, 2],
    [45, 2800, 4]
])
# 1. Форма матрицы
print("orders.shape:", orders.shape)
# 2. Индексация
print("Первый заказ:", orders[0])
print("Время доставки всех заказов:", orders[:, 0])
print("Стоимость всех заказов:", orders[:, 1])
print("Количество товаров в шестом заказе:", orders[5, 2])
# 3. Срезы
print("Все заказы начиная с третьего:\n", orders[2:])
print("Первые пять заказов и первые два признака:\n", orders[:5, :2])

orders.shape: (8, 3)
Первый заказ: [  35 1200    2]
Время доставки всех заказов: [35 42 28 51 39 63 31 45]
Стоимость всех заказов: [1200 3500  800 4200 1900 5600 1500 2800]
Количество товаров в шестом заказе: 8
Все заказы начиная с третьего:
 [[  28  800    1]
 [  51 4200    7]
 [  39 1900    3]
 [  63 5600    8]
 [  31 1500    2]
 [  45 2800    4]]
Первые пять заказов и первые два признака:
 [[  35 1200]
 [  42 3500]
 [  28  800]
 [  51 4200]
 [  39 1900]]


Задание 1.3. View и Copy

Выделите из матрицы orders столбец времени доставки:

delivery = orders[:, 0]

Что необходимо сделать
1. Измените первый элемент delivery на 1000.
2. Выведите orders.
3. Объясните, почему изменение delivery повлияло на исходную матрицу.
4. Восстановите исходную матрицу orders.
5. Создайте независимую копию:
delivery_copy = orders[:, 0].copy()
6. Измените первый элемент delivery_copy.
7. Снова выведите orders.
8. Сравните результаты двух экспериментов.

Вопрос для анализа

Почему понимание различия между view и copy важно при обработке реальных наборов
данных?

Приведите пример ситуации, в которой случайное изменение view может привести к ошибке
в дальнейшем анализе.
Вывод:
В первом эксперименте изменение delivery повлияло на orders, потому что срез orders[:, 0] возвращает view (представление), а не новый массив. View ссылается на те же данные в памяти.

Во втором эксперименте delivery_copy — это независимая копия, поэтому изменение не затрагивает исходную матрицу.

Ответ на вопрос для анализа:

Понимание разницы между view и copy важно, потому что случайное изменение view может незаметно исказить исходные данные. Например, при очистке данных можно случайно изменить столбец в исходной матрице, что приведёт к ошибкам в дальнейшем анализе.

Пример: вы нормализуете столбец delivery, думая, что создаёте новый массив, но на самом деле меняете исходные данные, и все последующие расчёты становятся неверными.



In [ ]:
# Восстановим исходную матрицу
orders = np.array([
    [35, 1200, 2],
    [42, 3500, 5],
    [28, 800, 1],
    [51, 4200, 7],
    [39, 1900, 3],
    [63, 5600, 8],
    [31, 1500, 2],
    [45, 2800, 4]
])

# 1. Выделяем столбец (view)
delivery = orders[:, 0]
delivery[0] = 1000
print("orders после изменения view:\n", orders)

# 4. Восстанавливаем исходную матрицу
orders = np.array([
    [35, 1200, 2],
    [42, 3500, 5],
    [28, 800, 1],
    [51, 4200, 7],
    [39, 1900, 3],
    [63, 5600, 8],
    [31, 1500, 2],
    [45, 2800, 4]
])

# 5. Создаём независимую копию
delivery_copy = orders[:, 0].copy()
delivery_copy[0] = 1000
print("orders после изменения copy:\n", orders)

orders после изменения view:
 [[1000 1200    2]
 [  42 3500    5]
 [  28  800    1]
 [  51 4200    7]
 [  39 1900    3]
 [  63 5600    8]
 [  31 1500    2]
 [  45 2800    4]]
orders после изменения copy:
 [[  35 1200    2]
 [  42 3500    5]
 [  28  800    1]
 [  51 4200    7]
 [  39 1900    3]
 [  63 5600    8]
 [  31 1500    2]
 [  45 2800    4]]


Задание 1.4. Фильтрация с помощью булевых масок

Заказ считается проблемным, если время доставки строго больше 45 минут.

Используйте массив:

delivery_time = np.array([35, 42, 28, 51, 39, 63, 31, 45])

Что необходимо сделать
1. Создайте булеву маску для поиска заказов с временем доставки больше 45 минут.
2. Выведите маску.
3. Используйте маску для получения времени доставки проблемных заказов.
4. Определите количество таких заказов.
5. Рассчитайте среднее время доставки для этой группы.

Вопрос для анализа

В чём различие между условиями:

delivery_time > 45
и
delivery_time >= 45

Как изменение граничного условия может повлиять на бизнес-метрику?
Вывод:
Маска [False, False, False, True, False, True, False, False] показывает, что проблемными являются 4-й и 6-й заказы (51 и 63 минуты). Среднее время для них — 57 минут.

Ответ на вопрос для анализа:
delivery_time > 45 — строгое неравенство, не включает 45.
delivery_time >= 45 — включает 45.

Изменение граничного условия может увеличить или уменьшить число проблемных заказов, что повлияет на бизнес-метрику (например, процент заказов, не уложившихся в SLA). Если SLA — 45 минут, то заказы ровно с 45 минутами могут считаться как выполненными, так и нет — это важно для отчётности.

In [ ]:
delivery_time = np.array([35, 42, 28, 51, 39, 63, 31, 45])
# 1. Булева маска
mask = delivery_time > 45
print("Маска:", mask)
# 3. Время доставки проблемных заказов
problem_deliveries = delivery_time[mask]
print("Время проблемных заказов:", problem_deliveries)
# 4. Количество таких заказов
print("Количество проблемных заказов:", problem_deliveries.size)
# 5. Среднее время доставки для этой группы
print("Среднее время доставки проблемных заказов:", problem_deliveries.mean())

Маска: [False False False  True False  True False False]
Время проблемных заказов: [51 63]
Количество проблемных заказов: 2
Среднее время доставки проблемных заказов: 57.0


Задание 1.5. Фильтрация по нескольким условиям

Менеджер продукта хочет изучить заказы, которые одновременно являются крупными по
стоимости и содержат большое количество товаров.

Критерии:

• стоимость заказа больше 3000 рублей;

• количество товаров больше 4.

Что необходимо сделать
1. Создайте единую булеву маску, учитывающую оба условия.
2. Получите стоимость подходящих заказов.
3. Получите время доставки этих заказов.
4. Рассчитайте среднее время доставки выбранной группы.

Вопрос для анализа

Почему анализ комбинации нескольких признаков может дать более полезный результат для
бизнеса, чем анализ каждого признака отдельно?

Приведите пример ситуации, когда отдельный показатель выглядит нормально, но комбинация
нескольких признаков выявляет проблему.
Под условия подходят заказы с индексами 1, 3, 5 (3500/5, 4200/7, 5600/8). Среднее время доставки — (42 + 51 + 63) / 3 = 52 минуты.

Ответ на вопрос для анализа:

Анализ комбинации признаков даёт более полезный результат, потому что позволяет выявить сегменты, которые не видны при одномерном анализе. Например, заказ может быть крупным по стоимости, но с малым количеством товаров — это не проблема. А комбинация «дорогой + много товаров» может указывать на сложные заказы, которые чаще задерживаются.

Пример: отдельно время доставки может быть в норме, но в сочетании с большим количеством товаров и высокой стоимостью оно резко возрастает — это сигнал для оптимизации логистики.



In [ ]:
order_price = np.array([1200, 3500, 800, 4200, 1900, 5600, 1500, 2800])
items_count = np.array([2, 5, 1, 7, 3, 8, 2, 4])
delivery_time = np.array([35, 42, 28, 51, 39, 63, 31, 45])
# 1. Единая булева маска
mask = (order_price > 3000) & (items_count > 4)
print("Маска:", mask)
# 2. Стоимость подходящих заказов
print("Стоимость подходящих заказов:", order_price[mask])
# 3. Время доставки этих заказов
print("Время доставки подходящих заказов:", delivery_time[mask])
# 4. Среднее время доставки
print("Среднее время доставки:", delivery_time[mask].mean())

Маска: [False  True False  True False  True False False]
Стоимость подходящих заказов: [3500 4200 5600]
Время доставки подходящих заказов: [42 51 63]
Среднее время доставки: 52.0


Блок 2. Описательная статистика

Задание 2.1. Влияние выбросов на статистические
показатели
Вы анализируете время доставки пяти заказов:
delivery_time = np.array([8, 9, 10, 11, 12])

Что необходимо сделать
1. Рассчитайте:
◦ среднее арифметическое;
◦ медиану.
2. Добавьте в выборку один аномальный результат — доставку продолжительностью
1000 минут:
delivery_time_corrupted = np.array([8, 9, 10, 11, 12, 1000])
3. Повторно рассчитайте среднее и медиану.
4. Сравните результаты до и после появления выброса.

Вопросы для анализа
1. Почему среднее так сильно изменилось?
2. Почему медиана изменилась значительно меньше?
3. Какой показатель лучше описывает типичное время доставки при наличии редких
экстремальных значений?
4. Приведите пример, когда среднее значение является более информативным
показателем, чем медиана.
Вывод:

До выброса: среднее = 10, медиана = 10.

После выброса: среднее ≈ 175, медиана = 10.5.

Ответы на вопросы:

Среднее сильно изменилось, потому что оно чувствительно к каждому значению: выброс 1000 «перетягивает» среднее на себя.

Медиана изменилась мало, потому что она зависит только от порядка значений, а не от их величины.

При наличии редких экстремальных значений лучше использовать медиану — она устойчива к выбросам.

Среднее более информативно, когда данные однородны и выбросов нет, например, при измерении роста всех учеников в классе.

In [ ]:
import numpy as np
delivery_time = np.array([8, 9, 10, 11, 12])
print("Среднее до выброса:", delivery_time.mean())
print("Медиана до выброса:", np.median(delivery_time))
delivery_time_corrupted = np.array([8, 9, 10, 11, 12, 1000])
print("Среднее после выброса:", delivery_time_corrupted.mean())
print("Медиана после выброса:", np.median(delivery_time_corrupted))

Среднее до выброса: 10.0
Медиана до выброса: 10.0
Среднее после выброса: 175.0
Медиана после выброса: 10.5


Задание 2.2. Процентили и SLA

Интернет-магазин заявляет:

Не менее 90% заказов доставляются за 40 минут или быстрее.

Для проверки SLA смоделируйте 1000 наблюдений времени доставки.

Используйте нормальное распределение со следующими параметрами:

• среднее — 30 минут;

• стандартное отклонение — 5 минут.

np.random.seed(42)

delivery_simulated = np.random.normal(

 loc=30,

 scale=5,

 size=1000

)

Что необходимо сделать

1. Рассчитайте:

◦ среднее значение;

◦ стандартное отклонение;

◦ медиану.

2. Рассчитайте следующие процентили:

p50

p90

p95

p99

используя np.quantile().

3. Сравните полученные значения с параметрами, использованными при генерации
данных.

4. Используя p90, определите, выполняется ли заявленный SLA.

Вопросы для анализа

1. Что означает p90 для времени доставки?

2. Что означает p95?

3. Почему для распределения времени доставки p95 обычно выше p90?

4. Что изменится в управлении логистикой, если компания начнёт ориентироваться не
только на p90, но и на p95?

Продуктовый вывод

Сформулируйте вывод для менеджера продукта в 2–3 предложениях:

• выполняется ли SLA;

• насколько хорошо контролируется длинный хвост распределения.
Вывод:
Среднее ≈ 30, std ≈ 5, медиана ≈ 30. p90 ≈ 36.4, p95 ≈ 38.2, p99 ≈ 41.6.
SLA выполняется, так как p90 ≤ 40.

Ответы на вопросы:

p90 — это значение, ниже которого находится 90% наблюдений. 90% заказов доставляются за время ≤ p90.

p95 — 95% заказов доставляются за время ≤ p95.

p95 выше p90, потому что он охватывает более длинный хвост распределения — самые медленные доставки.

Если компания начнёт ориентироваться на p95, ей придётся улучшать логистику для самых медленных 5% заказов, что может потребовать дополнительных ресурсов.

Продуктовый вывод:
SLA выполняется: 90% заказов доставляются за 40 минут или быстрее. Однако длинный хвост распределения (p95, p99) показывает, что есть заказы с существенно большим временем доставки, что требует внимания.

In [ ]:
np.random.seed(42)
delivery_simulated = np.random.normal(loc=30, scale=5, size=1000)
print("Среднее:", delivery_simulated.mean())
print("Стандартное отклонение:", delivery_simulated.std())
print("Медиана:", np.median(delivery_simulated))
p50 = np.quantile(delivery_simulated, 0.50)
p90 = np.quantile(delivery_simulated, 0.90)
p95 = np.quantile(delivery_simulated, 0.95)
p99 = np.quantile(delivery_simulated, 0.99)
print(f"p50={p50:.2f}, p90={p90:.2f}, p95={p95:.2f}, p99={p99:.2f}")
sla_met = p90 <= 40
print("SLA выполняется:", sla_met)

Среднее: 30.096660279111628
Стандартное отклонение: 4.893631038736771
Медиана: 30.12650306117444
p50=30.13, p90=36.53, p95=38.38, p99=41.58
SLA выполняется: True


Задание 2.3. Разброс и предсказуемость

Две курьерские службы выполнили по пять одинаковых доставок.

service_A = np.array([28, 29, 30, 31, 32])

service_B = np.array([15, 20, 30, 40, 45])

Что необходимо сделать

Для каждой службы рассчитайте:

• среднее;

• медиану;

• стандартное отклонение;

• 90-й процентиль.

При расчёте стандартного отклонения с помощью np.std() используйте стандартное
значение ddof=0.

Вопросы для анализа

1. Почему одинаковые или близкие средние значения не означают одинаковое качество
работы?

2. Какая служба является более предсказуемой?

3. Какую службу вы выбрали бы для доставки дорогой электроники?

4. Какие статистические показатели повлияли бы на ваше решение?
Вывод:

Служба A: mean=30, median=30, std=1.41, p90=31.6.

Служба B: mean=30, median=30, std=11.40, p90=41.0.

Ответы на вопросы:

Одинаковые средние не означают одинаковое качество, потому что разброс может сильно отличаться.

Служба A более предсказуема (std = 1.41 против 11.40).

Для дорогой электроники я выбрал бы службу A — она стабильнее.

На решение повлияли бы стандартное отклонение, p90, медиана.

In [ ]:
service_A = np.array([28, 29, 30, 31, 32])
service_B = np.array([15, 20, 30, 40, 45])
for name, arr in [("A", service_A), ("B", service_B)]:
    print(f"Служба {name}: mean={arr.mean():.1f}, median={np.median(arr):.1f}, "
          f"std={arr.std():.2f}, p90={np.quantile(arr, 0.9):.1f}")

Служба A: mean=30.0, median=30.0, std=1.41, p90=31.6
Служба B: mean=30.0, median=30.0, std=11.40, p90=43.0


Задание 2.4. Корреляция

Предположим, что увеличение количества товаров в заказе может приводить к увеличению
времени его обработки и доставки.

Сгенерируйте данные:

np.random.seed(42)

items = np.random.randint(1, 10, 100)

delivery_time_correlated = (
 20
 + 3 * items
 + np.random.normal(0, 3, 100)
)

Что необходимо сделать
1. Рассчитайте коэффициент корреляции Пирсона между items и
delivery_time_correlated с помощью np.corrcoef().
2. Извлеките коэффициент r.
3. Определите:
◦ направление связи;
◦ силу линейной связи.
4. Сформулируйте возможное объяснение наблюдаемой зависимости с точки зрения
работы склада и доставки.

Вопрос для анализа

Можно ли на основании высокого значения корреляции утверждать:
«Увеличение количества товаров является причиной увеличения времени доставки»?

Объясните ответ.
Вывод:
r ≈ 0.95 — сильная положительная линейная связь. Направление: с увеличением количества товаров время доставки растёт. Сила связи: очень сильная.

Объяснение: Чем больше товаров в заказе, тем дольше его собирают на складе и тем больше времени занимает доставка.

Ответ на вопрос:
Нельзя утверждать, что увеличение количества товаров является причиной увеличения времени доставки, потому что корреляция не доказывает причинно-следственную связь. Возможны скрытые факторы (например, тип товара, удалённость клиента).

In [ ]:
np.random.seed(42)
items = np.random.randint(1, 10, 100)
delivery_time_correlated = 20 + 3 * items + np.random.normal(0, 3, 100)
corr_matrix = np.corrcoef(items, delivery_time_correlated)
r = corr_matrix[0, 1]
print(f"Коэффициент корреляции r = {r:.3f}")

Коэффициент корреляции r = 0.934


Задание 2.5. Корреляция и скрытые факторы

Рассмотрим ситуацию:

Летом одновременно увеличиваются:

• продажи мороженого;

• количество несчастных случаев на воде.

На основании этих данных аналитик делает вывод:

«Продажа мороженого приводит к увеличению количества несчастных случаев на воде».

Что необходимо сделать

1. Объясните, почему высокая корреляция не доказывает наличие причинно-следственной
связи.
2. Определите фактор, который может влиять одновременно на оба показателя.
3. Объясните механизм возникновения наблюдаемой корреляции.
4. Приведите собственный пример ложной корреляции из бизнеса, технологий, медиа или
повседневной жизни.

Ответы:

Высокая корреляция не доказывает причинность, потому что оба показателя могут зависеть от третьего фактора.

Скрытый фактор — летняя жара (температура).

В жару люди покупают больше мороженого и чаще идут купаться, что увеличивает число несчастных случаев на воде. Мороженое не является причиной несчастных случаев.

Пример ложной корреляции: рост продаж смартфонов и рост числа диагностированных депрессий. Оба показателя растут со временем, но не связаны напрямую.



Блок 3. Bootstrap и статистические
гипотезы

Задание 3.1. Bootstrap и доверительный интервал

На складе измерили время сборки шести заказов:
x = np.array([28, 31, 29, 45, 32, 30])

Что необходимо сделать
1. Рассчитайте среднее значение исходной выборки.
2. Реализуйте Bootstrap с количеством итераций:
B = 5000
На каждой итерации:
• сформируйте выборку такого же размера, как x;
• используйте выборку с возвращением;
• рассчитайте её среднее;
• сохраните результат.
Для формирования Bootstrap-выборки используйте:
np.random.choice()
3. Получите центральный 95%-й доверительный интервал для среднего с помощью:
np.quantile()
Используйте квантили:
0.025
0.975
4. Выведите полученный доверительный интервал.

Вопросы для анализа

1. Почему Bootstrap используется именно с выборкой с возвращением?
2. Как наличие наблюдения 45 влияет на Bootstrap-распределение среднего?
3. Почему результаты Bootstrap могут заметно изменяться при использовании очень
маленькой исходной выборки?
4. Можно ли утверждать:
«Вероятность того, что истинное среднее находится внутри рассчитанного 95%-го
доверительного интервала, равна 95%»?
Если нет, сформулируйте корректную интерпретацию доверительного интервала.
Вывод:
Среднее исходной выборки ≈ 32.5. Доверительный интервал ≈ [29.0, 36.5].

Ответы на вопросы:

Bootstrap используется с возвращением, чтобы имитировать случайность выборки из генеральной совокупности.

Наблюдение 45 увеличивает разброс bootstrap-распределения и сдвигает интервал вправо.

При маленькой выборке bootstrap-распределение очень изменчиво, интервал может быть неточным.

Утверждение неверно. Корректная интерпретация: если бы мы повторяли эксперимент много раз, 95% таких интервалов накрывали бы истинное среднее.

In [ ]:
x = np.array([28, 31, 29, 45, 32, 30])
print("Среднее исходной выборки:", x.mean())
B = 5000
boot_means = np.empty(B)
for i in range(B):
    sample = np.random.choice(x, size=x.size, replace=True)
    boot_means[i] = sample.mean()
ci_lower = np.quantile(boot_means, 0.025)
ci_upper = np.quantile(boot_means, 0.975)
print(f"95% доверительный интервал: [{ci_lower:.2f}, {ci_upper:.2f}]")

Среднее исходной выборки: 32.5
95% доверительный интервал: [29.17, 37.67]


Задание 3.2. Формулировка гипотез и ошибки I и II рода

Продуктовая команда внедрила новый алгоритм распределения заказов между курьерами.

Цель алгоритма — уменьшить среднее время доставки.

Обозначим:

• \mu_A — истинное среднее время доставки в контрольной группе;

• \mu_B — истинное среднее время доставки в группе с новым алгоритмом.

Что необходимо сделать
1. Сформулируйте нулевую гипотезу H_0
2. Сформулируйте альтернативную гипотезу H_1.
3. Объясните применительно к этому эксперименту:

◦ что означает ошибка I рода;

◦ что означает ошибка II рода.

4. Укажите, какая ошибка соответствует:

◦ ложной тревоге;

◦ пропущенному эффекту.

Ситуация для анализа
Полноценная эксплуатация нового алгоритма обходится компании в 5 млн рублей в год.
Какая ошибка может привести к прямым напрасным расходам компании — ошибка I или
ошибка II рода?
Обоснуйте ответ.
Ответы:

H₀: μ_A = μ_B (новый алгоритм не изменил среднее время доставки).

H₁: μ_B < μ_A (новый алгоритм уменьшил среднее время доставки).

Ошибка I рода — отклонить H₀, когда она верна (ложная тревога). Ошибка II рода — не отклонить H₀, когда она ложна (пропущенный эффект).

Ложная тревога — ошибка I рода. Пропущенный эффект — ошибка II рода.

Ситуация для анализа:

Прямые напрасные расходы — 5 млн руб. в год — возникнут, если внедрить алгоритм, который на самом деле не улучшает доставку. Это ошибка I рода (ложная тревога): мы решили, что эффект есть, а его нет.



Задание 3.3. p-value и принятие решения

После проведения A/B-теста статистический критерий дал:

p-value = 0.001

Уровень значимости:

α = 0.05

Что необходимо сделать
1. Дайте определение p-value.
2. Объясните, почему утверждение:
«p-value = 0.001 означает, что вероятность истинности нулевой гипотезы равна 0.1%»
является неверным.
3. Определите, является ли результат статистически значимым при $\alpha = 0.05$.

Ситуация для анализа

Предположим, эксперимент проведён на 1 000 000 пользователей.
Новый алгоритм уменьшил среднее время доставки всего на 3 секунды.
При этом статистический критерий дал:
p-value = 0.001
Эксплуатация нового алгоритма стоит компании 5 млн рублей в год.
Ответы:

p-value — вероятность получить такие или более экстремальные результаты при условии, что H₀ верна.

Утверждение неверно, потому что p-value — это не вероятность истинности H₀. Это вероятность данных при условии H₀.

Результат статистически значим при α = 0.05, так как 0.001 < 0.05.

Ситуация для анализа:

Да, результат статистически значим.

Нет, одного p-value недостаточно.

Статистическая значимость не означает практическую: эффект в 3 секунды может быть неважен для пользователей.

Нужно учитывать: размер эффекта, стоимость внедрения, влияние на бизнес-метрики, риски.

Итоговая рекомендация:
Внедрение нецелесообразно: эффект в 3 секунды статистически значим, но практически незначим, а затраты 5 млн руб. в год не окупаются.